frontend/src/routes/index.tsx

```diff
@@ -1,22 +1,50 @@
 import React, { FC } from "react";
 import { Routes, Route } from "react-router-dom";
-import ProtectedRoute from "./ProtectedRoute";
+import { AuthorizedRoute } from "./ProtectedRoute";
 import projectRoutes from "./subRoutes/projectRoutes";
 import contractRoutes from "./subRoutes/contractRoutes";
 import adminRoutes from "./subRoutes/adminRoutes";
 import reportRoutes from "./subRoutes/reportRoutes";
-import { Home, Login, PageNotFound } from "../pages";
+import { Home, Login, PageNotFound, Users } from "../pages";
 import { Main } from "../components";
+import useAuthorization from "hooks/useAuthorization";
+import keycloak from "keycloak";
+import { ICurrentUser } from "types";
 
 const AppRouter: FC = () => {
+  const { currentUser } = useAuthorization(keycloak);
   return (
     <Routes>
-      <Route path="/" element={<ProtectedRoute component={Main} />}>
-        <Route index element={<Home />} />
-        {projectRoutes}
-        {contractRoutes}
-        {adminRoutes}
-        {reportRoutes}
+      <Route element={<Main />}>
+        <Route
+          path="/"
+          element={
+            <AuthorizedRoute
+              currentUserRole={(currentUser as unknown as ICurrentUser)?.role_id?.label}
```

Comment:
If currentUser in useAuthorization was defined as an ICurrentUser we wouldn't have to do this casting stuff here.